zebra/zebra_rib.c

```diff
@@ -1660,6 +1660,180 @@ static bool rib_compare_routes(const struct route_entry *re1,
 	return false;
 }
 
+/* Update the VRF ID for the nexthop group */
```

Comment:
this needs to be clearly documented. it should go this way, is this right?

every nexthop will have VRF_DEFAULT set. 
nexthops that have an ifindex get the vrf of that interface
nexthops without an ifindex get the vrf of the overall route object

so ... I think Donald's comment is correct too.